06week/dodgeball/dodgeball.js

```diff
@@ -0,0 +1,278 @@
+      placeBorn: "New Orleans, Louisiana"
+    },
+  ]
+  
+  // Classes and Constructors
+
+  class DodgeBallPlayer {
+    constructor(person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience){
+      this.id = person.id;
+      this.name = person.name;
+      this.canThrowBall = canThrowBall;
+      this.canDodgeBall = canDodgeBall;
+      this.hasPaid = hasPaid;
+      this.isHealthy = isHealthy;
+      this.yearsExperience = yearsExperience;
+      this.age = person.age;
+    }
+  }
+  
+  // class Drafted extends DodgeBallPlayer {
+  //   constructor (person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience, teamColor, mascott){
+  //     super (person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience) 
+  //     this.teamColor = teamColor;
+  //     this.mascott = mascott;
+  //   }
+  // }
+
+  class BlueTeammate extends DodgeBallPlayer {
+    constructor (person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience, teamColor, mascott) {
+      super (person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience) 
+      this.teamColor = "blue";
+      this.mascott = "Blues";
+    }
+  }
+
+  class RedTeammate extends DodgeBallPlayer {
+    constructor (person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience, teamColor, mascott){
+      super (person, canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience) 
+      this.teamColor = "red";
+      this.mascott = "Reds";
+    }
+  }
+  
+
+
+
+  // List the People that can be turned into players
+  // should list the person name and skill set
+  // along with a button with function to make a dodge ball player
+
+  function listPeopleChoices() {
+
+
+    arrOfPeople.map(person => {
+      // Grab the people id that should display the lsit of people 
+      let listElement = document.getElementById('people');
+
+      // display the name and skill
+      let li = document.createElement("li");
+      li.appendChild(document.createTextNode(person.name + " - " + person.skillSet+"  "));
+
+      // create and append a "make player" button
+      let makePlayer = document.createElement("button");
+      makePlayer.innerHTML = "Commit to Draft";
+      makePlayer.addEventListener('click', function() {interestedPlayer(person)} );
+      li.appendChild(makePlayer);
+      
+      // Append the entire line, name, skill, red button, blue button
+      listElement.appendChild(li);
+    })
+  }
+
+
+
+
+  // List Dodge Ball Players
+  // should list the person name 
+  // along with two buttons with function to make a red team or blue team
+  // uses the person.id to push player the dodge ball payers list
+  // should add the class New DodgeBallPlayer to arrOfPeople
+  let dodgeBallPlayer = [];
+
+  function interestedPlayer(person) {
+
+      dodgeBallPlayer.push(new DodgeBallPlayer(person));
+      let playerElement = document.getElementById('players');
+      
+      dodgeBallPlayer.map(person => {
```

Comment:
you dont need to do this work on every person in the list of dodgeBallPlayers, just the person you just created.

Reply:
excellent comment, I was copying and translating code from an above function without full comprehension. Fixed.